cmd/bosun/expr/funcs.go

```diff
@@ -657,7 +673,11 @@ func timeGraphiteRequest(e *State, T miniprofiler.Timer, req *graphite.Request)
 		}
 		var val interface{}
 		val, err = e.cache.Get(key, getFn)
-		resp = val.(graphite.Response)
+		// hand out copy of the value, so callers can't accidentally modify data in cache
+		if val, ok := val.(*graphite.Response); ok && val != nil {
```

Comment:
didn't we have that whole discussion where we concluded processing functions such as avg() can modify result sets in place and so we need to protect by always copying out of the cache?